### PR TITLE
Node/Gov: Fix panic in public RPC

### PR DESCRIPTION
**Pull request by bruce-riley**

### CODE REVIEW

**Review by SEJeff — APPROVED** [review]

File: node/pkg/governor/governor_monitoring.go [hunk 1]
```diff
@@ -325,6 +325,10 @@ func (gov *ChainGovernor) IsVAAEnqueued(msgId *publicrpcv1.MessageID) (bool, err
 	gov.mutex.Lock()
 	defer gov.mutex.Unlock()
 
+	if msgId == nil {
+		return false, fmt.Errorf("no message ID specified")
+	}
+
```
Comment (claudijd) [inline node/pkg/governor/governor_monitoring.go:331]: Should EmitterChain similarly be nil checked?

File: node/pkg/governor/governor_monitoring_test.go [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+	"go.uber.org/zap"
+)
+
+func TestIsVAAEnqueuedNilMessageID(t *testing.T) {
+	logger, _ := zap.NewProduction()
+	gov := NewChainGovernor(logger, nil, GoTestMode)
+	enqueued, err := gov.IsVAAEnqueued(nil)
+	require.EqualError(t, err, "no message ID specified")
+	assert.Equal(t, false, enqueued)
+}
```
Comment (claudijd) [inline node/pkg/governor/governor_monitoring_test.go:17]: Can we add a test where msgId is supplied with a nil EmitterChain?

File: node/pkg/publicrpc/publicrpcserver.go [hunk 3]
```diff
@@ -157,6 +157,10 @@ func (s *PublicrpcServer) GovernorGetEnqueuedVAAs(ctx context.Context, req *publ
 func (s *PublicrpcServer) GovernorIsVAAEnqueued(ctx context.Context, req *publicrpcv1.GovernorIsVAAEnqueuedRequest) (*publicrpcv1.GovernorIsVAAEnqueuedResponse, error) {
 	resp := &publicrpcv1.GovernorIsVAAEnqueuedResponse{}
 
+	if req.MessageId == nil {
+		return nil, status.Error(codes.InvalidArgument, "no message ID specified")
+	}
```
Comment (claudijd) [inline node/pkg/publicrpc/publicrpcserver.go:162]: Can we have an additional nil check for EmitterID being nil?  I suspect if you call .Number() on nil you will also get a nil ptr deref.

File: node/pkg/publicrpc/publicrpcserver_test.go [hunk 4]
```diff
@@ -63,3 +64,19 @@ func TestGetSignedVAABadAddress(t *testing.T) {
...
+
+	// A message without the messageId set should not panic but return an error instead.
+	msg := publicrpcv1.GovernorIsVAAEnqueuedRequest{}
+	assert.NotPanics(t, func() {
+		_, err := server.GovernorIsVAAEnqueued(ctx, &msg)
+		assert.Error(t, err)
+		expected_err := status.Error(codes.InvalidArgument, "no message ID specified")
+		assert.Equal(t, expected_err, err)
+	})
+}
```
Comment (claudijd) [inline node/pkg/publicrpc/publicrpcserver_test.go:82]: Add a test that included a nil Emitter

**Review by claudijd — CHANGES_REQUESTED** [review]
Add tests/checks for nil emitter

File: node/pkg/governor/governor_monitoring.go (hunk 1 above)
Comment (claudijd) [reply node/pkg/governor/governor_monitoring.go:331]: Discussed offline, this will default to zero if unset, which is not a nil ptr deref issue.

**Review by claudijd — APPROVED** [review]
